### ISSUE TITLE
Update proto.hrc to use ".tpl" for "Smarty PHP templates", not ".smarty"

### ISSUE THREAD

**Issue by unxed** [colorer/Colorer-schemes#76]
Smarty itself does not define extension for template files strictly, but ".tpl" is standard de facto, and it is also used by IntelliJ PhpStorm by default.

I guess Turbo Prolog for .tpl is used significantly less often.

UPD:

sublime smarty plugin assumes .tpl:
https://github.com/amitsnyderman/sublime-smarty

atom plugin uses .tpl:
https://atom.io/packages/language-smarty

vscode plugin for smarty allows both, but .tpl goes first:
"extensions": [".tpl",".smarty",".html.smarty"],
https://github.com/jens1o/vscode-smarty/blob/master/package.json

**Comment by AnrDaemon** [comment]
`.tpl` is too generic for a specific coloring extension.
If your Smarty files are always .tpl, and you are not using this extension for anything else, you can always use autoload to override the behavior.
Something akin to
```
<?xml version="1.0" encoding='UTF-8'?>
<!DOCTYPE hrc PUBLIC
  "-//Cail Lomecb//DTD Colorer HRC take5//EN"
  "http://colorer.sf.net/2003/hrc.dtd"
>
<hrc version="take5" xmlns="http://colorer.sf.net/2003/hrc"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://colorer.sf.net/2003/hrc http://colorer.sf.net/2003/hrc.xsd"
>

<annotation><documentation>
  Smarty extension override.
</documentation></annotation>

<prototype name="Smarty.tpl" group="scripts" description="Use .tpl as Smarty extension">
  <filename>/\.(tpl)$/i</filename>
</prototype>

<type name="Smarty.tpl">

  <import type="Smarty"/>

  <scheme name="Smarty.tpl">
    <inherit scheme="Smarty"/>
  </scheme>

</type>
</hrc>
```

**Comment by AnrDaemon** [comment]
Back to https://github.com/colorer/Colorer-schemes/issues/75#issuecomment-495887235

**Comment by unxed** [comment]
@AnrDaemon,
>  what about treating as smarty templates all `.tpl` files with first line starting with `{` or `<` symbols?

**Comment by AnrDaemon** [comment]
You can do it in your autoloaded `.hrc`. All features available in catalog are available in autoloaded files.
Just add something like `<firstline>/^\s*([{&lt;])/xi</firstline>` to the prototype defintion.

**Comment by unxed** [comment]
> You can do it in your autoloaded `.hrc`. All features available in catalog are available in autoloaded files. Just add something like `<firstline>/^\s*([{&lt;])/xi</firstline>` to the prototype defintion.

Wrote my custom user.hrc with a trick you mentioned, but it still not working. What am I doing wrong?

```
<?xml version="1.0" encoding='UTF-8'?>
<!DOCTYPE hrc PUBLIC
  "-//Cail Lomecb//DTD Colorer HRC take5//EN"
  "http://colorer.sf.net/2003/hrc.dtd">
<hrc version="take5" xmlns="http://colorer.sf.net/2003/hrc"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://colorer.sf.net/2003/hrc http://colorer.sf.net/2003/hrc.xsd">

  <prototype name="smarty" group="inet" description="Smarty PHP Templates">
    <firstline>/^\s*([{&lt;])/xi</firstline>
    <location link="/usr/lib/far2l/Plugins/colorer/base/hrc/inet/smarty.hrc"/>
    <filename>/\.tpl$/</filename>
  </prototype>
</hrc>
```

**Comment by unxed** [comment]
Just for history reasons: fixed in 1ef8c8cfc2487a91fff300111638dc5f82f94ddc
